src/Processors/Executors/PipelineExecutor.cpp

```diff
@@ -46,6 +44,16 @@ PipelineExecutor::PipelineExecutor(Processors & processors, QueryStatus * elem)
 
         throw;
     }
+    if (process_list_element)
+    {
+        auto settings = process_list_element->context.lock()->getSettings();
+        limits.max_execution_time = settings.max_execution_time;
+        overflow_mode = settings.timeout_overflow_mode;
```

Comment:
I think it may be more reasonable to add timeout check logic into `process_list_element`.

Reply:
Makes sense to me. I'll add a `checkTimeLimit()` to `process_list_element` and use it from the PipeLineExecutor